docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,444 @@
+
+In the following example connecting `incoming.alignedChild` to `outgoing.alignedChild`, whether `incoming.alignedChild` is aligned with `incoming` is irrelevant because the `:<>=` only computes alignment relative to the thing being connected to, and `incoming.alignedChild` is aligned with `incoming.alignedChild`.
+
+```scala mdoc:silent
+class Example1a extends RawModule {
+  val incoming = IO(Flipped(new MixedAlignmentBundle))
+  val outgoing = IO(new MixedAlignmentBundle)
+  outgoing.alignedChild :<>= incoming.alignedChild // whether incoming.alignedChild is aligned/flipped to incoming is IRRELEVANT to what gets connected with :<>=
+}
+```
+
+While `incoming.flippedChild`'s alignment with `incoming` does not affect our operators, it does influence whether `incoming.flippedChild` is an output or input port of my module.
+A common source of confusion is to mistake the process for determining whether `incoming.flippedChild` is an output/input (the port-direction computation) with the process for determing how `:<>=` connects who to who (the connection-direction computation).
+While both processes consider relative alignment, they are distinct.
+
+The port-direction computation always computes alignment relative to the component marked with `IO`. An `IO(Flipped(gen))` is an input port, and any member of `gen` that is aligned/flipped with `gen` is an input/output port. An `IO(gen)` is an output port, and any member of `gen` that is aligned/flipped with `gen` is an output/input port.
+
+The connection-direction computation always computes alignment based on explicit consumer/producer referenced for the connection. If one connects `incoming :<>= outgoing`, alignments are computed based on `incoming` and `outgoing`. If I connect `incoming.alignedChild :<>= outgoing.alignedChild`, then alignments are computed based on `incoming.alignedChild` and `outgoing.alignedChild` (and the alignment of `incoming` to `incoming.a` is irrelevant).
+
+This means that users can try to assign to input ports of their module! If I write `x :<>= y`, and `x` is an input to the current module, then that is what the connection is trying to do. However, because input ports are not assignable from within the current module, Chisel will throw an error. This is the same error a user would get using a mono-directioned operator: `x := y` will throw the same error if `x` is an input module. *Whether a component is assignable is irrelevant to the semantics of any connection operator assigning to it.*
+
+In summary, the port-direction computation is relative to the root marked `IO`, but connection-direction computation is relative to the consumer/producer that the connection is doing. This has the positive property that connection semantics are solely based on the Chisel types of the consumer/producer (nothing more, nothing less).
```

Comment:
> solely based on the Chisel types of the consumer/producer (nothing more, nothing less).

this section makes me think Chisel type DOES include the alignment, but then again i'm unclear if you intend to include that in "chisel type equivalence" or not

Reply:
Hopefully my distinction between structural type equivalence and alignment type equivalence clears this up?

Reply:
sorry, no, it doesnt help because you don't mention structural tyep or alignment type here, you jsut say chisel type and i never heard chisel type relationship to these other 2 terms is